internal/events/submanager.go

```diff
@@ -467,17 +469,42 @@ func (s *subscriptionMGR) addSubscription(spec *eventsapi.SubscriptionInfo) erro
 	if spec.Filter.EventFilter == "" && spec.Filter.ChaincodeId != "" {
 		spec.Filter.EventFilter = ".*"
 	}
+
+	// A subscription is based on an event client and a registration. We must ensure that
+	// on restart all subscriptions can be restored. We must avoid allowing different subscriptions
+	// to register for the same parameters against the same event client, which would fail. NOTE that
+	// upon restart, all subscriptions will have been set for the "fromBlock" to the latest checkpoint,
+	// which would likely be the same across the board. As a result, "fromBlock" can not be a key segment
+	// to index event clients.
+	//
+	// Because of the above, we require that a subscription must have a unique composite key made up of:
+	// - channel ID
+	// - chaincode ID (empty string if creating a block event listener)
+	// - block type (endorser or config)
+	// - event filter
+	//
+	// Note that, because of the way event clients are cached by fabric-sdk-go, subscriptions having the
+	// same channel ID and chaincode ID, but different the event filter will share the same event client.
+	// This means subsequent subscriptions will NOT get historical events, because the offset in the event
+	// client will have been set to the latest block.
```

Comment:
If I have a chaincode that emits two different events, and has been emitting both of them for some number of blocks already, does this mean I can only get the history from block `0` for one of the events?

Reply:
that unfortunately is the state of things as of now. To fix that we'd need to make another update to the fabric-sdk-go code

Reply:
That just seems fundamentally broken in the SDK